.pre-commit-config.yaml

```diff
@@ -0,0 +1,51 @@
+
+---
+repos:
+  - repo: git://github.com/Lucas-C/pre-commit-hooks
+    rev: v1.1.7
+    hooks:
+      - id: remove-tabs
+
+  - repo: git://github.com/pre-commit/pre-commit-hooks
+    rev: v3.1.0
+    hooks:
+      - id: trailing-whitespace
+      - id: check-merge-conflict
+      - id: end-of-file-fixer
+      - id: check-added-large-files
+      - id: check-byte-order-marker
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: check-json
+      - id: check-symlinks
+      - id: detect-private-key
+      - id: check-ast
+      - id: debug-statements
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.1.0
+    hooks:
+      - id: check-toml
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.782
+    hooks:
+      - id: mypy
+        exclude: '^(docs|tasks|tests)|setup\.py'
+        args: [--ignore-missing-imports]
+
+  - repo: https://github.com/psf/black
+    rev: 19.10b0
+    hooks:
+      - id: black
+
+  - repo: https://gitlab.com/PyCQA/flake8
+    rev: 3.8.3
+    hooks:
+      - id: flake8
+        additional_dependencies: ['pep8-naming']
+        # Ignore all format-related checks as Black takes care of those.
+        args: ['--ignore', 'E2,W5,F401,N802', '--select', 'E,W,F,N', '--max-line-length=136']
```

Comment:
@fridex I ignored the following error codes `F401`, `N802` and removed the below part from `pre-commit-config.yaml`
```
- repo: git://github.com/pycqa/pydocstyle.git
    rev: 5.0.2
    hooks:
      - id: pydocstyle
```
Is it OK?
They gave a lot of errors like `missing docstrings` and `lowercasing most of the function names`. I can modify them, I'm afraid they maybe used somewhere and changing them might break something, what do you think I should do `ignore` or `work on those errors`?

Reply:
F401 and N802 sound reasonable to have to me. Any specific reason why they have been removed?

Open a PR with a standard pre-commit config we have, I can take a look and add docstrings if it requires my action.

Reply:
@fridex , I added standard `pre-commit-config.yaml` from https://github.com/thoth-station/template-project, https://github.com/thoth-station/invectio/pull/93/commits/998dd10774ea248803b8aff629dd955137466509